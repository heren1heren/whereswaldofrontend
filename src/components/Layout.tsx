import { Link } from 'react-router-dom';
import { LayoutType } from '../typeDeclaration';

export const Layout = ({
  children,

  title = 'Where is waldo',
}: LayoutType) => {
  return (
    <div className=" d-flex  flex-column justify-content-between vh-100 text-center">
      <h1>{title}</h1>
      <nav className="nav justify-content-center  ">
        <Link to="/records" className="nav-link ">
          {' '}
          Record
        </Link>

        <Link to="/" className="nav-link ">
          {' '}
          Home Page
        </Link>
      </nav>
      <main className=" flex-grow-1 p-lg-4"> {children}</main>
      <footer className=" text-center bg-black text-white">
        {' '}
        Made By
        <a
          href="https://github.com/heren1heren/whereswaldofrontend"
          className=" ms-md-1  text-decoration-none"
        >
          Cookie Turtle
        </a>
      </footer>
    </div>
  );
};
