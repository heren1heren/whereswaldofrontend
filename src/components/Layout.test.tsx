import { Layout } from './Layout';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
describe('initilizing test', () => {
  // render screen
  // expect inside document
  it('should work with title props ', () => {
    const component = render(
      <MemoryRouter>
        <Layout title={'test'}>hello</Layout>
      </MemoryRouter>
    ); //<MemoryRouter> should be used in order to test Link contained component
    expect(screen.getByText('test')).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });
  // expect return ... with corresponding props
});
