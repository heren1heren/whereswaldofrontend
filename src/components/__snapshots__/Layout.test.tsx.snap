// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`initilizing test > should work with title props  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" d-flex  flex-column justify-content-between vh-100 text-center"
      >
        <h1>
          test
        </h1>
        <nav
          class="nav justify-content-center  "
        >
          <a
            class="nav-link "
            href="/records"
          >
             
            Record
          </a>
          <a
            class="nav-link "
            href="/"
          >
             
            Home Page
          </a>
        </nav>
        <main
          class=" flex-grow-1 p-lg-4"
        >
           
          hello
        </main>
        <footer
          class=" text-center bg-black text-white"
        >
           
          Made By
          <a
            class=" ms-md-1  text-decoration-none"
            href="https://github.com/heren1heren/whereswaldofrontend"
          >
            Cookie Turtle
          </a>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" d-flex  flex-column justify-content-between vh-100 text-center"
    >
      <h1>
        test
      </h1>
      <nav
        class="nav justify-content-center  "
      >
        <a
          class="nav-link "
          href="/records"
        >
           
          Record
        </a>
        <a
          class="nav-link "
          href="/"
        >
           
          Home Page
        </a>
      </nav>
      <main
        class=" flex-grow-1 p-lg-4"
      >
         
        hello
      </main>
      <footer
        class=" text-center bg-black text-white"
      >
         
        Made By
        <a
          class=" ms-md-1  text-decoration-none"
          href="https://github.com/heren1heren/whereswaldofrontend"
        >
          Cookie Turtle
        </a>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
